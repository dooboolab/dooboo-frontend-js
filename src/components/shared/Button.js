// @flow
import React, { Component } from 'react';
import styled from 'styled-components';
import { WhiteButton, DarkButton } from '../ui/Buttons';

type Props = {
  id?: string,
  white?: boolean,
  imgSrc?: any,
  srcset?: any,
  txt?: string,
  onClick: () => void,
};

type State = {

};

const Text = styled.span`
  font-size: 14px;
  color: rgb(128, 109, 216);
`;

const LogoImg = styled.img`
  position: absolute;
  left: 16px;
  height: 20px;
  width: 20px;
  object-fit: cover
`;

export class Button extends Component<Props, State> {
  static defaultProps: Props = {
    onClick: () => {},
  };

  render() {
    const { white, onClick, imgSrc, txt, srcset } = this.props;
    if (this.props.white) {
      return (
        <WhiteButton
          onClick={() => onClick()}
        >
          {
            this.props.imgSrc
              ? <LogoImg
                src={this.props.imgSrc}
                srcSet={this.props.srcset}
              />
              : null
          }
          <Text>{this.props.txt}</Text>
        </WhiteButton>
      );
    }
    return (
      <DarkButton
        onClick={() => onClick()}
      >
        {
          this.props.imgSrc
            ? <LogoImg
              src={this.props.imgSrc}
              srcSet={this.props.srcset}
            />
            : null
        }
        <Text>{this.props.txt}</Text>
      </DarkButton>
    );
  }
}

export default Button;
