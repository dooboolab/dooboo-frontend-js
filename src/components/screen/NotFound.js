// @flow
import React, { Component } from 'react';
import Button from '../shared/Button';
import styled from 'styled-components';

const Container = styled.div`
  display: 'flex',
  justify-content: 'center',
`;

type Props = {
  history: any,
};

type State = {

};

class NotFound extends Component<Props, State> {
  render() {
    const { history } = this.props;
    return (
      <Container>
        <Button
          id='btn'
          onClick={() => history.goBack()}
          txt='back to tab page'
        />
      </Container>
    );
  }
}

export default NotFound;
