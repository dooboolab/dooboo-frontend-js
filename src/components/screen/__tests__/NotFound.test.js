import React from 'react';
import renderer from 'react-test-renderer';

import NotFound from '../NotFound';
import Button from '../../shared/Button';

// test for the container page in dom
describe('NotFound page DOM rendering test', () => {
  let tree;
  const component = <NotFound {...props} />;

  it('component and snapshot matches', () => {
    tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const props = {
  history: {
    goBack: jest.fn(),
  },
};

describe('Interaction', () => {
  let rendered;
  let instance;
  const component = <NotFound {...props} />;

  beforeAll(() => {
    rendered = renderer.create(component);
    instance = rendered.root;
  });

  it('Simulate onClick', () => {
    // const spy = jest.spyOn(rendered.getInstance(), 'onClick');
    const button = instance.findByType(Button);
    button.props.onClick();
    expect(props.history.goBack).toBeCalled();
  });
});
