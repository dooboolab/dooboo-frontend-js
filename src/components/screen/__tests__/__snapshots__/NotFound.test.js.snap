// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound page DOM rendering test component and snapshot matches 1`] = `
<div
  className="sc-ifAKCX cyMJPX"
>
  <button
    className="sc-bdVaJa erJscN"
    onClick={[Function]}
  >
    <span
      className="sc-htpNat gUfFlb"
    >
      back to tab page
    </span>
  </button>
</div>
`;
